extensions/wikia/LinkSuggest/LinkSuggest.class.php

```diff
@@ -172,19 +173,42 @@ static function getLinkSuggest(WebRequest $request) {
 			 * It uses fact that page titles can't start with lowercase letter.
 			 */
 			$pageTitlePrefilter = "";
-			if( strlen($queryLower) >= 2 ) {
-				$pageTitlePrefilter = "(
-							( convert(binary convert(page_title using latin1) using utf8) " . $db->buildLike(strtoupper($queryLower[0]) . strtolower($queryLower[1]) , $db->anyString() ) . " ) OR
-							( convert(binary convert(page_title using latin1) using utf8) " . $db->buildLike(strtoupper($queryLower[0]) . strtoupper($queryLower[1]) , $db->anyString() ) . " ) ) AND ";
-			} else if( strlen($queryLower) >= 1 ) {
-				$pageTitlePrefilter = "( page_title " . $db->buildLike(strtoupper($queryLower[0]) , $db->anyString() ) . " ) AND ";
+
+			// use mb_substring to get the first & second chars
+			// in case of multi-byte
+			$firstChar = mb_substr($queryLower, 0, 1);
+			$secondChar = mb_substr($queryLower, 1, 1);
+
+			if( mb_strlen( $queryLower ) >= 2 ) {
+				if ( LinkSuggest::containsMultibyteCharacters( $firstChar . $secondChar ) ) {
+					$pageTitlePrefilter = "(
```

Comment:
Yes I've checked. As we talked about it earlier, because of the db connection settings the MB string is munged so the latin conversion is required. It took a bit of SQL sleuthing to find out why when I copy and pasted the SQL from PHP into MySQL it ran fine but when I ran from the app it failed.